### PR TITLE
Dvc 8454 home view

### PR DESCRIPTION
**Pull request by nsalamad**
# Changes

image

* add Home section
  * logout button
  * Organization selector (TODO: unset project and display empty state for variable usages - DVC-8557)
  * Project selector
  * `getAllProjects` now call `projects list` from the CLI since only project keys are needed
* update webpack config to load components from webview-ui-toolkit package

## TODO
* Write integration tests

### CODE REVIEW

File: media/vscode.css [hunk 1]
```diff
@@ -100,3 +106,171 @@ input::placeholder,
...
+  width: fit-content;
+  border-radius: .3rem;
+  padding: .3rem;
+}
+
+.icon-button:hover {
+  background-color: var(--vscode-button-secondaryHoverBackground);
+}
+
+/* Home section
```
Comment (laurawarr) [inline media/vscode.css:140]: we could move this into a style file specific to the home section

File: src/cli/ProjectsCLIController.ts [hunk 2]
```diff
@@ -4,32 +4,38 @@ import { BaseCLIController } from './BaseCLIController'
...
+    return repoConfig.project
+  }
+
   public async getAllProjects() {
     const projects = StateManager.getFolderState(this.folder.name, KEYS.PROJECTS)
     if (projects) {
       return projects
     }
-    const { code, error, output } = await this.execDvc('projects get')
+    const { code, error, output } = await this.execDvc('projects list')
```
Comment (laurawarr) [inline src/cli/ProjectsCLIController.ts:28]: why the change from fetching/storing the full object to only using the key?

File: src/cli/ProjectsCLIController.ts [hunk 3]
```diff
@@ -4,32 +4,38 @@ import { BaseCLIController } from './BaseCLIController'
...
+    if (stateProject) {
+      return stateProject
+    }
+    const repoConfig = await getRepoConfig(this.folder)
+    StateManager.setFolderState(this.folder.name, KEYS.PROJECT_ID, repoConfig.project)
+    return repoConfig.project
+  }
+
   public async getAllProjects() {
     const projects = StateManager.getFolderState(this.folder.name, KEYS.PROJECTS)
```
Comment (laurawarr) [inline src/cli/ProjectsCLIController.ts:24]: If we're changing the structure of data that's stored here, we'll need to make sure that this handles parsing both the old and new data

File: src/cli/OrganizationsCLIController.ts [hunk 4]
```diff
@@ -13,30 +13,58 @@ export type Organization = {
 
 export class OrganizationsCLIController extends BaseCLIController {
   projectController: ProjectsCLIController
+  baseCLIController: BaseCLIController
```
Comment (laurawarr) [inline src/cli/OrganizationsCLIController.ts:16]: the OrganizationsCLIController is extending the BaseCLIController, so it shouldn't need to create an instance of it as well

File: src/cli/ProjectsCLIController.ts (hunk 2 above)
Comment (nsalamad) [reply src/cli/ProjectsCLIController.ts:28]: I'm actually unsure about this. Emma made these changes when she did the initial work. I figured there was a reason for it but I can't seem to find one myself. I'll can change it back and see if anything breaks

File: src/cli/ProjectsCLIController.ts [hunk 5]
```diff
@@ -4,32 +4,38 @@ import { BaseCLIController } from './BaseCLIController'
...
   key: string
 }
 
 export class ProjectsCLIController extends BaseCLIController {
+  public async getActiveProject() {
+    const stateProject = StateManager.getFolderState(this.folder.name, KEYS.PROJECT_ID)
+    if (stateProject) {
+      return stateProject
+    }
+    const repoConfig = await getRepoConfig(this.folder)
```
Comment (laurawarr) [inline src/cli/ProjectsCLIController.ts:18]: It's possible that the repo config is not defined or doesn't contain a project, would this handle that case?

Comment (nsalamad) [reply src/cli/ProjectsCLIController.ts:18]: it does not. I think if the repo config does not exist then the project in the user config should be used. That's guranteed to be set since we force the user to set a project on login

**Review by jsalaber — APPROVED** [review]

File: src/StateManager.ts [hunk 6]
```diff
@@ -76,6 +77,7 @@ export class StateManager {
   static setFolderState(folder: string, key: KEYS.FEATURE_CONFIGURATIONS, value: Record<string, FeatureConfiguration[]> | undefined): Thenable<void>
   static setFolderState(folder: string, key: KEYS.ENVIRONMENTS, value: Record<string, Environment> | undefined): Thenable<void>
   static setFolderState(folder: string, key: KEYS.ORGANIZATION, value: Organization | undefined): Thenable<void>
+  static setFolderState(folder: string, key: KEYS.ORGANIZATIONS, value: Record<string, Organization> | undefined): Thenable<void>
```
Comment (laurawarr) [inline src/StateManager.ts:80]: the organizations wouldn't need to be scoped by folder, since the orgs available to the current user will be the same for all folders

File: src/cli/OrganizationsCLIController.ts [hunk 7]
```diff
@@ -19,24 +19,62 @@ export class OrganizationsCLIController extends BaseCLIController {
     this.projectController = new ProjectsCLIController(folder)
   }
 
+  public async getActiveOrganization() {
+    const stateOrg = StateManager.getFolderState(this.folder.name, KEYS.ORGANIZATION)
+    if (stateOrg) {
+      return stateOrg
+    }
+    // dvc status returns org name from the repo config if it exists, otherwise it returns from user config
+    const orgName = (await this.status()).organization
```
Comment (laurawarr) [inline src/cli/OrganizationsCLIController.ts:28]: when this method is called we're just using the name, we could possibly just return the name and save the cli call to fetch all orgs

Comment (nsalamad) [reply src/cli/OrganizationsCLIController.ts:28]: the dropdown uses display name for the label. But i can just display the name instead

Comment (nsalamad) [reply src/cli/OrganizationsCLIController.ts:28]: although in the dashboard we use the display name for orgs in the dropdown. Might be better to use the display name in the extension as well for consistency

Comment (nsalamad) [reply src/cli/OrganizationsCLIController.ts:28]: `display_name` is being used as the label and `name` is used for value of the each dropdown option [here](https://github.com/DevCycleHQ/vscode-extension/pull/102/files#diff-10a83d3464ffa423233be698b840a117a3fe0b71fcfd5813e0688a7961d29840R84). The fallback for the label is `name`. We need the call to `getAllOrganizations` to fetch all orgs so that we can have entire objects with display names

Comment (nsalamad) [reply src/cli/OrganizationsCLIController.ts:28]: only reason i want to use the display name is cause adam brought up a good point last week about how we should refer to things the same way on the dashboard and integrations so that people don't get confused

Comment (laurawarr) [reply src/cli/OrganizationsCLIController.ts:28]: Yep, we should definitely use the display name — but that's sourced from getAllOrganizations (not the active organizations). We need to call `getAllOrganizations` when building the view, but we don't necessarily need to call it to get the active organization name

Comment (nsalamad) [reply src/cli/OrganizationsCLIController.ts:28]: ahhh okay i see what you're saying. `getActiveOrganization` doesn't need to exist, we should just be able to get the active org from state every time as it should always be set. Will fix

**Review by laurawarr — APPROVED** [review]
